habitat-lab/habitat/tasks/rearrange/actions/actions.py

```diff
@@ -461,227 +461,87 @@ def step(self, ee_pos, **kwargs):
-        self.base_vel_ctrl = habitat_sim.physics.VelocityControl()
-        self.base_vel_ctrl.controlling_lin_vel = True
-        self.base_vel_ctrl.lin_vel_is_local = True
-        self.base_vel_ctrl.controlling_ang_vel = True
-        self.base_vel_ctrl.ang_vel_is_local = True
-        # Initialize the controller
-        max_lin_speed = 30  # real Stretch max linear velocity
-        max_ang_speed = 20.94  # real Stretch max angular velocity
-        lin_speed_gain = 10000.0  # the linear gain
-        ang_speed_gain = 10000.0  # the angular gain
-        lin_tol = 0.001  # 5cm linear error tol
-        ang_tol = 0.001  # 5cm angular error tol
-        max_heading_ang = np.pi / 10.0
-        self.controller = Controller(
-            v_max=max_lin_speed,
-            w_max=max_ang_speed,
-            lin_gain=lin_speed_gain,
-            ang_gain=ang_speed_gain,
-            lin_error_tol=lin_tol,
-            ang_error_tol=ang_tol,
-            max_heading_ang=max_heading_ang,
-        )
+        self._allow_back = config.allow_back
+        self._collision_threshold = config.collision_threshold
+        self._navmesh_offset = config.navmesh_offset
+        self._allow_dyn_slide = config.allow_dyn_slide
+        self._min_displacement = (
+            config.min_displacement
+        )  # minimum displacement
+        self._max_displacement_along_axis = config.max_displacement_along_axis
+        self._max_turn_radians = (
+            config.max_turn_degrees * np.pi / 180
+        )  # maximum turn waypoint
+        self._min_turn_radians = (
+            config.min_turn_degrees * np.pi / 180
+        )  # minimum turn waypoint
+        self._allow_lateral_movement = config.allow_lateral_movement
+        self._allow_simultaneous_turn = config.allow_simultaneous_turn
+
+    def collision_check(self, trans, target_trans):
+        """
+        trans: the transformation of the current location of the robot
+        target_trans: the transformation of the target location of the robot given the center original Navmesh
+        """
+        # Get the offset positions
+        num_check_cylinder = len(self._navmesh_offset)
+        nav_pos_3d = [
+            np.array([xz[0], xz[1], 0.0]) for xz in self._navmesh_offset
+        ]
+        cur_pos = [trans.transform_point(xyz) for xyz in nav_pos_3d]
+        goal_pos = [target_trans.transform_point(xyz) for xyz in nav_pos_3d]
+
+        # For step filter of offset positions
+        end_pos = []
+        for i in range(num_check_cylinder):
+            pos = self._sim.step_filter(cur_pos[i], goal_pos[i])
+            # Sanitize the height
+            pos[1] = 0.0
+            cur_pos[i][1] = 0.0
+            goal_pos[i][1] = 0.0
+            end_pos.append(pos)
+
+        # Planar move distance clamped by NavMesh
+        move = []
+        for i in range(num_check_cylinder):
+            move.append((end_pos[i] - goal_pos[i]).length())
+
+        # For detection of linear or angualr velocities
```

Comment:
```suggestion
        # For detection of linear or angular velocities
```

Reply:
Unclear what you mean by "For detection of linear or angular velocities"

Reply:
Not sure. Didn't notice while copying from [here](https://github.com/facebookresearch/habitat-lab/blob/6f7c6b9c906a60b104a31ca3ad7051dcaed5e0f9/habitat-lab/habitat/tasks/rearrange/actions/actions.py#L549). Removed this now.